examples/depth-filters/rs-depth-filters.cpp

```diff
@@ -0,0 +1,68 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2017 Intel Corporation. All Rights Reserved.
+
+#include <librealsense2/rs.hpp> // Include RealSense Cross Platform API
+#include "example.hpp"          // Include short list of convenience functions for rendering
+#include "proc/synthetic-stream.h"
+#include "proc/decimation-filter.h"
+#include "proc/spatial-filter.h"
+
+// Capture Example demonstrates how to
+// capture depth and color video streams and render them to the screen
+int main(int argc, char * argv[]) try
+{
+    // Create a simple OpenGL window for rendering:
+    window app(1280, 720, "RealSense Capture Example");
+    // Declare two textures on the GPU, one for color and one for depth
+    texture depth_image, filtered_image;
+
+    // Declare depth colorizer for pretty visualization of depth data
+    rs2::colorizer color_map;
+    //rs2::decimation_filter depth_filter;
+    rs2::spatial_filter depth_filter;
```

Comment:
Please add some descriptive comments here on what are `rs2::spatial_filter` and `rs2::option_range`.